includes/discovery/functions.inc.php

```diff
@@ -577,37 +577,39 @@ function discover_link($local_port_id, $protocol, $remote_port_id, $remote_hostn
 
 function discover_storage(&$valid, $device, $index, $type, $mib, $descr, $size, $units, $used = null)
 {
-    d_echo("Discover Storage: $index, $type, $mib, $descr, $size, $units, $used\n");
-
-    if ($descr && $size > '0') {
-        $storage = dbFetchRow('SELECT * FROM `storage` WHERE `storage_index` = ? AND `device_id` = ? AND `storage_mib` = ?', array($index, $device['device_id'], $mib));
-        if ($storage === false || !count($storage)) {
-            $insert = dbInsert(
-                array(
-                    'device_id' => $device['device_id'],
-                    'storage_descr' => $descr,
-                    'storage_index' => $index,
-                    'storage_mib' => $mib,
-                    'storage_type' => $type,
-                    'storage_units' => $units,
-                    'storage_size' => $size,
-                    'storage_used' => $used,
-                ),
-                'storage'
-            );
+    if (ignore_storage($device['os'], $descr) != 1) {
```

Comment:
Can you change this to be like this?
```php
if (ignore_storage($device['os'], $descr)) {
    return;
}
```
It saves an indent and long if statement.